document-store/src/main/java/org/hypertrace/core/documentstore/mongo/MongoCollection.java

```diff
@@ -131,6 +132,7 @@ private BasicDBObject prepareUpsert(Key key, Document document) throws JsonProce
     JsonNode sanitizedJsonNode = recursiveClone(jsonNode, this::encodeKey);
     BasicDBObject setObject = BasicDBObject.parse(MAPPER.writeValueAsString(sanitizedJsonNode));
     setObject.put(ID_KEY, key.toString());
+    setObject.put(LAST_UPDATED_TIME, System.currentTimeMillis());
```

Comment:
Can we add java doc on top of public function `upsert` that it adds two special keys on top of input document _ID and `lastUpdatedTime`?

Reply:
ok, will add